src/types/message.rs

```diff
@@ -116,75 +183,137 @@ pub enum ForwardKind {
     },
 }
 
-#[derive(Debug, Deserialize, PartialEq, Clone)]
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 #[serde(untagged)]
 pub enum MediaKind {
     Animation {
+        /// Message is an animation, information about the animation. For
+        /// backward compatibility, when this field is set, the document field
+        /// will also be set.
         animation: Animation,
+
         #[doc(hidden)]
-        /// "For backward compatibility" (c) Telegram Docs
+        /// "For backward compatibility" (c) Telegram Docs.
         #[serde(skip)]
         document: (),
+
+        /// Caption for the animation, 0-1024 characters.
         caption: Option<String>,
+
+        /// For messages with a caption, special entities like usernames, URLs,
+        /// bot commands, etc. that appear in the caption.
         #[serde(default = "Vec::new")]
         caption_entities: Vec<MessageEntity>,
     },
     Audio {
+        /// Message is an audio file, information about the file.
         audio: Audio,
+
+        /// Caption for the audio, 0-1024 characters.
         caption: Option<String>,
+
+        /// For messages with a caption, special entities like usernames, URLs,
+        /// bot commands, etc. that appear in the caption.
         #[serde(default = "Vec::new")]
         caption_entities: Vec<MessageEntity>,
     },
     Contact {
+        /// Message is a shared contact, information about the contact.
         contact: Contact,
     },
     Document {
+        /// Message is a general file, information about the file.
         document: Document,
+
+        /// Caption for the document, 0-1024 characters.
         caption: Option<String>,
+
+        /// For messages with a caption, special entities like usernames, URLs,
+        /// bot commands, etc. that appear in the caption.
         #[serde(default = "Vec::new")]
         caption_entities: Vec<MessageEntity>,
     },
     Game {
+        /// Message is a game, information about the game. [More about games
+        /// »].
```

Comment:
That's weird formatting...

Reply:
`#[rustfmt::skip]`? It's weird too!

Reply:
You can just add a line break after `...about the game. `

Reply:
Ok.